sdk/devcenter/azure-developer-devcenter/src/main/java/com/azure/developer/devcenter/EnvironmentsClientBuilder.java

```diff
@@ -0,0 +1,338 @@
+        this.devCenter = devCenter;
+        return this;
+    }
+
+    /*
+     * The DNS suffix used as the base for all devcenter requests.
+     */
+    @Generated private String devCenterDnsSuffix;
+
+    /**
+     * Sets The DNS suffix used as the base for all devcenter requests.
+     *
+     * @param devCenterDnsSuffix the devCenterDnsSuffix value.
+     * @return the EnvironmentsClientBuilder.
+     */
+    @Generated
+    public EnvironmentsClientBuilder devCenterDnsSuffix(String devCenterDnsSuffix) {
+        this.devCenterDnsSuffix = devCenterDnsSuffix;
+        return this;
+    }
+
+    /*
+     * Service version
+     */
+    @Generated private DevCenterServiceVersion serviceVersion;
+
+    /**
+     * Sets Service version.
+     *
+     * @param serviceVersion the serviceVersion value.
+     * @return the EnvironmentsClientBuilder.
+     */
+    @Generated
+    public EnvironmentsClientBuilder serviceVersion(DevCenterServiceVersion serviceVersion) {
+        this.serviceVersion = serviceVersion;
+        return this;
+    }
+
+    /*
+     * The retry policy that will attempt to retry failed requests, if applicable.
+     */
+    @Generated private RetryPolicy retryPolicy;
+
+    /**
+     * Sets The retry policy that will attempt to retry failed requests, if applicable.
+     *
+     * @param retryPolicy the retryPolicy value.
+     * @return the EnvironmentsClientBuilder.
+     */
+    @Generated
+    public EnvironmentsClientBuilder retryPolicy(RetryPolicy retryPolicy) {
+        this.retryPolicy = retryPolicy;
+        return this;
+    }
+
+    /**
+     * Builds an instance of DevCenterClientImpl with the provided parameters.
+     *
+     * @return an instance of DevCenterClientImpl.
+     */
+    @Generated
+    private DevCenterClientImpl buildInnerClient() {
+        HttpPipeline localPipeline = (pipeline != null) ? pipeline : createHttpPipeline();
+        String localDevCenterDnsSuffix = (devCenterDnsSuffix != null) ? devCenterDnsSuffix : "devcenter.azure.com";
+        DevCenterServiceVersion localServiceVersion =
+                (serviceVersion != null) ? serviceVersion : DevCenterServiceVersion.getLatest();
+        DevCenterClientImpl client =
+                new DevCenterClientImpl(
+                        localPipeline,
+                        JacksonAdapter.createDefaultSerializerAdapter(),
+                        tenantId,
+                        devCenter,
+                        localDevCenterDnsSuffix,
+                        localServiceVersion);
+        return client;
+    }
+
+    @Generated
+    private HttpPipeline createHttpPipeline() {
+        Configuration buildConfiguration =
+                (configuration == null) ? Configuration.getGlobalConfiguration() : configuration;
+        if (httpLogOptions == null) {
+            httpLogOptions = new HttpLogOptions();
+        }
+        if (clientOptions == null) {
+            clientOptions = new ClientOptions();
+        }
```

Comment:
For @weidongxu-microsoft, these should be updated to use local variables instead of mutating the builder

Reply:
https://github.com/Azure/autorest.java/issues/1768

Yeah, these are ancient code that I've not really pay attention when reviewing. Thx for pointing out.